auto_submit/lib/service/validation_service.dart

```diff
@@ -128,40 +145,154 @@ class ValidationService {
+
+        await removeLabelAndComment(
+            githubService: gitHubService,
+            repositorySlug: slug,
+            prNumber: prNumber,
+            prLabel: Config.kAutosubmitLabel,
+            message: message);
+
+        log.info(message);
+
         shouldReturn = true;
       }
     }
+
     if (shouldReturn) {
       log.info('The pr ${slug.fullName}/$prNumber with message: $ackId should be acknowledged.');
       await pubsub.acknowledge('auto-submit-queue-sub', ackId);
       log.info('The pr ${slug.fullName}/$prNumber is not feasible for merge and message: $ackId is acknowledged.');
       return;
     }
+
     // If PR has some failures to ignore temporarily do nothing and continue.
     for (ValidationResult result in results) {
       if (!result.result && result.action == Action.IGNORE_TEMPORARILY) {
         return;
       }
     }
+
     // If we got to this point it means we are ready to submit the PR.
-    bool processed = await processMerge(config, result, messagePullRequest);
-    if (processed) await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+    ProcessMergeResult processed =
+        await processMerge(config: config, queryResult: result, messagePullRequest: messagePullRequest);
+
+    if (!processed.result) {
+      String message = 'auto label is removed for ${slug.fullName}, pr: $prNumber, ${processed.message}.';
+
+      await removeLabelAndComment(
+          githubService: gitHubService,
+          repositorySlug: slug,
+          prNumber: prNumber,
+          prLabel: Config.kAutosubmitLabel,
+          message: message);
+
+      log.info(message);
+    }
+
     log.info('Ack the processed message : $ackId.');
+    await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+  }
+
+  /// The logic for processing a revert request and opening the follow up
+  /// review issue in github.
+  Future<void> processRevertRequest({
+    required Config config,
+    required QueryResult result,
+    required github.PullRequest messagePullRequest,
+    required String ackId,
+    required PubSub pubsub,
+  }) async {
+    ValidationResult revertValidationResult = await revertValidation!.validate(result, messagePullRequest);
+
+    github.RepositorySlug slug = messagePullRequest.base!.repo!.slug();
+    final int prNumber = messagePullRequest.number!;
+    final GithubService gitHubService = await config.createGithubService(slug);
+
+    if (revertValidationResult.result) {
+      // Approve the pull request automatically as it has been validated.
+      await approverService!.revertApproval(result, messagePullRequest);
+
+      ProcessMergeResult processed = await processMerge(
+        config: config,
+        queryResult: result,
+        messagePullRequest: messagePullRequest,
+      );
+
+      if (processed.result) {
+        try {
+          github.Issue issue = await gitHubService.createIssue(
+            repositorySlug: github.RepositorySlug('flutter', 'flutter'),
```

Comment:
nit: Leave a comment here stating issues are only created in flutter/flutter.

Long term, this may be something we need to create a config for if we plan to expand usage outside the Flutter org.

Reply:
Done.